src/components/popUp/popUpSrv.js

```diff
@@ -0,0 +1,156 @@
+
+            PopUpSrv.closePopup = function(reject,reason){
+                if(!reason){
+                    reason = 'closed';
+                }
+                popUpsPlaceHolderElement.empty();
+                if (popupInstance.scope) {
+                    popupInstance.scope.$destroy();
+                }
+                popupDefer[(reject ? 'reject' : 'resolve')](reason);
+            };
+
+            PopUpSrv.popup = function popup(wrapperCls,header,body,buttonsArr){
+                //kill current popup if exists
+                if(popupInstance){
+                    PopUpSrv.closePopup();
+                }
+                var childScope = $rootScope.$new(true);
+                childScope.d = {};
+
+                popupDefer = $q.defer();
+                popupInstance = {};
+                popupInstance.promise = popupDefer.promise;
+
+                var template =
+                    '<div class="%wrapperCls%">' +
+                    '<div class="znk-popup-wrapper">' +
+                    '<div class="znk-popup-header">%header%</div>' +
+                    '<div class="znk-popup-body">%body%</div>' +
+                    '<div class="znk-popup-buttons">' +
+                    '<div ng-repeat="button in d.buttons" class="button-wrapper">' +
+                    '<button class="btn" ng-click="d.btnClick(button)" ng-class="button.type" analytics-on="click" analytics-event="click-popup-{{button.text}}" analytics-category="popup">{{button.text}}</button>' +
+                    '</div>' +
+                    '</div>' +
+                    '</div>';
+
+                wrapperCls = wrapperCls ? ' ' + wrapperCls : '';
+                template = template.replace('%wrapperCls%',wrapperCls);
+
+                header = header || '';
+                template = template.replace('%header%',header);
+
+                body = body || '';
+                template = template.replace('%body%',body);
+
+                if(angular.isDefined(buttonsArr) && !angular.isArray(buttonsArr)){
+                    buttonsArr = [buttonsArr];
+                }
+                childScope.d.buttons = buttonsArr;
+                childScope.d.btnClick = function(button){
+                    if(button.hasOwnProperty('rejectVal')){
+                        childScope.d.close(button.rejectVal,true);
+                    }else{
+                        childScope.d.close(button.resolveVal);
+                    }
+                };
+
+                childScope.d.close = function(reason,reject){
+                    var animationLeaveProm = $animate.leave($template);
+                    animationLeaveProm.then(function(){
+                        if(childScope){
+                            childScope.$destroy();
+                            childScope = null;
+                        }
+                        popupInstance = null;
+                        if (angular.isDefined(popupDefer) && popupDefer !== null) {
```

Comment:
this condition is returning multiple times

maybe we should add 
angular.isUndefinedOrNull = function(val) {
    return angular.isUndefined(val) || val === null 
}